packages/shared/src/utils/cfd/cfd.ts

```diff
@@ -23,7 +25,20 @@ const CFD_text = {
 // sub_account_type: "financial" | "financial_stp" | "swap_free"
 // *
 // sub_account_type financial_stp only happens in "financial" market_type
-export const getCFDAccountKey = ({ market_type, sub_account_type, platform, shortcode }) => {
+type TPlatform = 'dxtrade' | 'mt5';
+type TMarketType = 'financial' | 'synthetic' | 'gaming' | undefined;
+type TShortcode = 'svg' | 'bvi' | 'labuan' | 'vanuatu';
+type TGetAccount = {
+    market_type: TMarketType;
+    sub_account_type?: 'financial' | 'financial_stp' | 'swap_free' | undefined;
+    platform: TPlatform;
```

Comment:
maybe we better should set a default value for example to empty string, then to set as a type 'undefinied'?
'?' is 'undefinied', so we better should check if the value is passed, then to use 'undefinied' in types.
how do you think?

Reply:
for `sub_account_type` I removed the undefined and it is working properly!